File: src/main/java/net/guizhanss/fastmachines/libs/guizhanlib/Scheduler.java
package net.guizhanss.fastmachines.libs.guizhanlib;

import org.bukkit.Bukkit;
import org.bukkit.plugin.Plugin;

/**
 * A tiny task scheduler. Java-8-safe port of GuizhanLib's {@code Scheduler}.
 *
 * @author Mooy1, ybw0014 (original), ported for the Slimefun5 fork
 */
public class Scheduler {

    private final Plugin plugin;

    public Scheduler(Plugin plugin) {
        this.plugin = plugin;
    }

    public void run(Runnable runnable) {
        Bukkit.getScheduler().runTask(plugin, runnable);
    }

    public void runAsync(Runnable runnable) {
        Bukkit.getScheduler().runTaskAsynchronously(plugin, runnable);
    }

    public void repeat(int intervalTicks, Runnable runnable) {
        repeat(intervalTicks, 1, runnable);
    }

    public void repeatAsync(int intervalTicks, Runnable runnable) {
        repeatAsync(intervalTicks, 1, runnable);
    }

    public void repeat(int intervalTicks, int delayTicks, Runnable runnable) {
        Bukkit.getScheduler().runTaskTimer(plugin, runnable, delayTicks, Math.max(1, intervalTicks));
    }

    public void repeatAsync(int intervalTicks, int delayTicks, Runnable runnable) {
        Bukkit.getScheduler().runTaskTimerAsynchronously(plugin, runnable, delayTicks, Math.max(1, intervalTicks));
    }
}

File: src/main/kotlin/net/guizhanss/fastmachines/libs/guizhanlib/Scheduler.kt
package net.guizhanss.fastmachines.libs.guizhanlib

import org.bukkit.Bukkit
import org.bukkit.plugin.Plugin
import kotlin.math.max

/**
 * A tiny task scheduler. Java-8-safe, Kotlin port of GuizhanLib's `Scheduler` - see the package-level
 * note in [net.guizhanss.fastmachines.libs.guizhanlib].
 *
 * @author Mooy1, ybw0014 (original), ported for the Slimefun5 fork
 */
class Scheduler(private val plugin: Plugin) {

    fun run(runnable: Runnable) {
        Bukkit.getScheduler().runTask(plugin, runnable)
    }

    fun runAsync(runnable: Runnable) {
        Bukkit.getScheduler().runTaskAsynchronously(plugin, runnable)
    }

    fun repeat(intervalTicks: Int, runnable: Runnable) {
        repeat(intervalTicks, 1, runnable)
    }

    fun repeatAsync(intervalTicks: Int, runnable: Runnable) {
        repeatAsync(intervalTicks, 1, runnable)
    }

    fun repeat(intervalTicks: Int, delayTicks: Int, runnable: Runnable) {
        Bukkit.getScheduler().runTaskTimer(plugin, runnable, delayTicks.toLong(), max(1, intervalTicks).toLong())
    }

    fun repeatAsync(intervalTicks: Int, delayTicks: Int, runnable: Runnable) {
        Bukkit.getScheduler().runTaskTimerAsynchronously(plugin, runnable, delayTicks.toLong(), max(1, intervalTicks).toLong())
    }
}
